### DISCUSSION TITLE
sorting docker-compose file

### DISCUSSION THREAD

**Discussion by lzinga** [mikefarah/yq#1132] [Q&A] [answered]
I am setting up a workspace that will eventually generate my full docker-compose.yml files. I got yq to merge files properly but am wondering if there is a way to sort based on the keys in a specific order.

For example - have the keys sort first by version, then services and put networks and volumes at the bottom in a manner that compose files typically look like. Right now I am using this command to merge the files `yq ea '. as $item ireduce ({}; . * $item )' myfiles -I4`

**Comment by mikefarah** [comment, marked as answer]
yq will keep the keys in the order of the first file given, so you should be able to do something like:

```
yq ea '. as $item ireduce ({}; . * $item )' template.yml myfiles -I4
```

where template.yml is:
```yaml
version:
services:
networks:
volumes:
```
